core-bundle/src/EventListener/VirtualFieldsMappingListener.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsHook;
+use Contao\DataContainer;
+use Contao\DC_Table;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+
+/**
+ * Adds "saveTo" automatically to fields without an "sql" definition. Defines
+ * "saveTo" targets automatically as "virtualTarget". Adds an "sql" definition
+ * automatically to virtual field targets.
+ */
+#[AsHook('loadDataContainer')]
+class VirtualFieldsMappingListener
+{
+    public function __construct(private readonly string $defaultStorageName = 'jsonData')
+    {
+    }
+
+    public function __invoke(string $table): void
```

Comment:
Again here, this whole stuff should use the `DcaExtracter` and pre-cached data.

Reply:
See above, the `DcaExtractor` only extracts the SQL data.